### PR TITLE
Add organizations

### PR DESCRIPTION
**Pull request by RachelWyatt**
### Description

This PR adds Organizations to the data structure. The motivation for this is extending the project to the Pinto Abalone organization (and hopefully others in the future!). 

An Organization ```has_many``` Facilities and Users. Facilities and Users, in turn, each ```belong_to``` an Organization. 

This allows for separation of data -- Pinto should not be able to see Facilities or Reports for White Abalone and vice versa.

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

* Test suite passes with updated factories

### CODE REVIEW

File: app/controllers/facilities_controller.rb [hunk 1]
```diff
@@ -69,6 +69,6 @@ def set_facility
 
     # Never trust parameters from the scary internet, only allow the white list through.
     def facility_params
-      params.require(:facility).permit(:name, :code)
+      params.require(:facility).permit(:name, :code, :organization_id)
```
Comment (mdworken) [inline app/controllers/facilities_controller.rb:72]: I don't think we want to add this to the params - can't we just calculate it from current_user?

File: app/controllers/facilities_controller.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class FacilitiesController < ApplicationController
   # GET /facilities
   # GET /facilities.json
   def index
-    @facilities = Facility.all
+    @facilities = Facility.where(organization_id: current_user.organization_id)
```
Comment (jtu0) [inline app/controllers/facilities_controller.rb:7]: How do we feel about always scoping facility to the current user's organization?

I know some people haaaaaaaaaaate default scopes, but I wonder if it wouldn't be a good idea here.

@bhaibel what do you think?

File: app/models/organization.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Organization < ApplicationRecord
+    has_many :users
+    has_many :facilities
+    has_many :tanks, through: :facilities
```
Comment (jtu0) [inline app/models/organization.rb:4]: I _think_ we don't need the `has_many :tanks`, but feel like my understanding of the stakeholder needs is a little too weak to say for sure.

**Comment by jtu0** [comment]
@RachelWyatt want any help getting this PR merged? happy to pair if you need it, or jump in and help on anything you want.

**Comment by RachelWyatt** [comment]
@jtu0 I'll pull down some time this weekend, fix the conflicts and get it merged in! I may slack you if I run into any issues

**Pull request merged**